### PR TITLE
fixed callback for fetch engines

### LINKED ISSUE #963: fix callback for fetch engines
see #961

### PR DESCRIPTION
**Pull request by plata**
Tested with a renamed copy of Wine (in JS).

fixes #963

### CODE REVIEW

File: phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java [hunk 1]
```diff
@@ -49,20 +49,22 @@ public EnginesSource(ScriptInterpreter scriptInterpreter, ObjectMapper objectMap
...
+                                        .withName(engineName)
+                                        .withDescription(engineName)
+                                        .withSubCategories(unSerialize(output))
+                                        .build();
+                                engines.add(engineCategoryDTO);
                             }, this::throwError),
                     this::throwError);
         }
-
+        callback.accept(engines);
```
Comment (madoar) [inline phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java]: Is it always the case that the callback method is called **after** the `interactiveScriptSession` eval calls have been successfully terminated?

Comment (plata) [reply phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java]: You are right. It doesn't work like this.

Comment (madoar) [reply phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java]: Maybe we should add a synchronous way to evaluate a script?
Afterwards we could synchronously execute the scripts and call the callback afterwards.
In addition we could wrap all these steps in a single lambda which we execute in a new thread.

Comment (plata) [reply phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java]: I have a different idea. Let me try something.

Comment (plata) [reply phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java]: Ok. It works. I'm not sure if there is a better way than calling this function like I did it now.

**Comment by plata** [comment]
I was thinking about this. But I decided not to do it on purpose because it's not really functionality. Maybe it is somehow possible to do this without an explicit function.

**Pull request merged**